pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_controller.go

```diff
@@ -95,6 +91,29 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req kube_ctrl.Request
 	return kube_ctrl.Result{}, nil
 }
 
+func (r *GatewayReconciler) createOrUpdateInstance(ctx context.Context, gateway *gatewayapi.Gateway) (*mesh_k8s.GatewayInstance, error) {
+	instance := &mesh_k8s.GatewayInstance{
+		ObjectMeta: kube_meta.ObjectMeta{
+			Namespace: gateway.Namespace,
+			Name:      gateway.Name,
```

Comment:
Should use `GenerateName` here (and really also for the Kuma Gateway that we create).

Reply:
Gateway is a little tricky since we create it via the store interface (though I suspect we could probably create it directly via the k8s API)

Reply:
Yeh, I think that creating the Kuma Gateway in the Kubernetes API is probably a better choice, since that keeps the gatewayapi controller separate from Kuma infrastructure (per my proposal that it should be a separate controller).

Reply:
https://github.com/kumahq/kuma/issues/3517